bases/PlaneBase.py

```diff
@@ -0,0 +1,41 @@
+"""
+
+Usage example:
+
+    >>> win = pygame.display.set_mode((500, 500))
+    >>> plane = PlaneBase(win)
+
+"""
+
+import abc
+
+import pygame
+
+from interfaces.IShape import IShape
+from interfaces.IWorld import IWorld
+
+
+class PlaneBase(IWorld):
+
+    def __init__(
+        self,
+        surface: pygame.Surface,
+        defaultColor: pygame.Color,
+        gridSizeX: int,
+        gridSizeY: int,
+        gridSizeZ: int = None,
+    ):
+        self._shapes: list[IShape] = []
+        self.win: pygame.Surface = surface
+        self.color = defaultColor
+        self.gridSizeX: int = gridSizeX
+        self.gridSizeY: int = gridSizeY
+        if gridSizeZ is not None:
+            self.gridSizeZ = gridSizeZ
+
+    @abc.abstractmethod
+    def draw(self):
+        """Draw the world/plane to the given surface"""
+
+    def addShape(self, shape: IShape):
```

Comment:
This i puzzling, either should change to:
```suggestion
    def add(self, shape: IShape):
```
Or the interface method should rename to `addShape`

Reply:
Done, with a few changes